src/main.rs

```diff
@@ -44,7 +47,17 @@ fn main() -> ExitCode {
 }
 
 fn run() -> Result<(), Box<dyn std::error::Error>> {
-    let ctx = Context::init()?;
+    let mut ctx = Context::init()?;
+
+    if ctx.stdin {
```

Comment:
I moved it over to `Context::init`, however for my taste it was a bit more elegant to modify the `Context` struct, i.e.:

https://github.com/solidiquis/erdtree/blob/a618a7a770d59500b75f6b71a1ef9e8374147ce0/src/main.rs#L59

instead of modifying the `args` Vec, i.e.:

https://github.com/solidiquis/erdtree/blob/a4ccb7f40f3898bf395f4be2db1558f8ca428349/src/render/context/mod.rs#L153-L155

Reply:
Sounds great to me!